qiskit/providers/ibmq/credentials/hubgroupproject.py

```diff
@@ -16,5 +16,94 @@
 
 from collections import namedtuple
 
-HubGroupProject = namedtuple('HubGroupProject',
-                             ['hub', 'group', 'project'])
+from .exceptions import HubGroupProjectInvalidStateError
+
+HubGroupProjectTuple = namedtuple('HubGroupProjectTuple',
+                                  ['hub', 'group', 'project'])
```

Comment:
Decided to keep an internal nametuple that is to be used only by this class. I thought it would make the code more legible, since it would allow us to spec types such as `HubGroupProjectTuple` instead of `Tuple[str, str, str]`. Also, it would allow us to index the individual tuple fields as well.